File: extract.js
const fs = require('fs');

let text = fs.readFileSync('index.html', 'utf-8');
text = text.replace(/<style[^>]*>[\s\S]*?<\/style>/gi, '');
text = text.replace(/<script[^>]*>[\s\S]*?<\/script>/gi, '');
text = text.replace(/<!--[\s\S]*?-->/g, '');
text = text.replace(/<[^>]+>/g, '\n');
text = text.replace(/\n\s*\n\s*\n/g, '\n\n');

console.log(text.trim());

File: extract.py
import re

with open('index.html', 'r', encoding='utf-8') as f:
    text = f.read()

text = re.sub(r'<style.*?>.*?</style>', '', text, flags=re.DOTALL)
text = re.sub(r'<script.*?>.*?</script>', '', text, flags=re.DOTALL)
text = re.sub(r'<!--.*?-->', '', text, flags=re.DOTALL)
text = re.sub(r'<[^>]+>', '\n', text)
text = re.sub(r'\n\s*\n\s*\n', '\n\n', text)

print(text.strip())
